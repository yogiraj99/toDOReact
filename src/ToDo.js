import React, {Component} from 'react';
import ToDoItem from "./ToDoItem";

class ToDo extends Component {
  constructor(props) {
    super(props);
    this.state = props.state;
    this.updateToDoItem = this.updateToDoItem.bind(this);
    this.updateStatus = this.updateStatus.bind(this);
    this.getAllToDoItems = this.getAllToDoItems.bind(this);
  }

  render() {
    return (
        <div className="toDo">
          <h1>{this.getTitle()}</h1>
          <ul>{this.getAllToDoItems()}</ul>
        </div>)
  }

  updateToDoItem(toDoItem) {
    let toDoItems = this.state.toDoItems;
    let oldToDoItem = toDoItems.find((item) => item.id === toDoItem.id);
    let toDoItemIndex = toDoItems.indexOf(oldToDoItem);
    toDoItems[toDoItemIndex] = toDoItem;
    this.setState({toDoItems:toDoItems});
    this.updateStatus();
  }

  getAllToDoItems() {
    let toDoItems = this.props.state.toDoItems;
    return toDoItems.map((item) => {
      return (<li key={`todoItem${item.id}`}>
        <ToDoItem state={item} update={this.updateToDoItem}/>
      </li>)
    })
  }

  updateStatus() {
    this.setState({status: false});
    if (this.areAllTasksDone()) {
      this.setState({status: true});
    }
  }

  areAllTasksDone() {
    return this.state.toDoItems.every((item) => item.status);
  }

  getTitle() {
    if (this.state.status) {
      return <del>{this.state.title}</del>;
    }
    return this.state.title;
  }
}

export default ToDo
