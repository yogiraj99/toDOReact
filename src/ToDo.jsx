import React, {Component} from 'react';
import ToDoItem from "./ToDoItem";

class ToDo extends Component {
  render() {
    return (<div>
      {this.props.state.toDoItems.map((i)=><ToDoItem state={i} key={`todoItem${i.id}`}/>)}
    </div>)
  }
}

export default ToDo
