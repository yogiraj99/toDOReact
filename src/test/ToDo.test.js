import React from 'react';
import ToDo from '../ToDo';
import renderer from 'react-test-renderer';

test('Should have toDo item provided in list', () => {
  let state={toDoItems:[{task:"Wake up early", status:true,id:20100}]};
  const component = renderer.create(
      <ToDo state={state}/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Should have all toDo items provided in list', () => {
  let state={toDoItems:[{task:"Wake up early", status:true,id:20100},{task:"Sleep early", status:true,id:20101}]};
  const component = renderer.create(
      <ToDo state={state}/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});