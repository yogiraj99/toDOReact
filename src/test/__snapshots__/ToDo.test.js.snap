// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should have all toDo items provided in list 1`] = `
<div
  className="toDo"
>
  <h1 />
  <ul>
    <li>
      <div
        className="toDoItem"
      >
        <h4
          className="task"
          onClick={[Function]}
        >
          <del>
            Wake up early
          </del>
        </h4>
      </div>
    </li>
    <li>
      <div
        className="toDoItem"
      >
        <h4
          className="task"
          onClick={[Function]}
        >
          <del>
            Sleep early
          </del>
        </h4>
      </div>
    </li>
  </ul>
</div>
`;

exports[`Should have toDo item provided in list 1`] = `
<div
  className="toDo"
>
  <h1 />
  <ul>
    <li>
      <div
        className="toDoItem"
      >
        <h4
          className="task"
          onClick={[Function]}
        >
          <del>
            Wake up early
          </del>
        </h4>
      </div>
    </li>
  </ul>
</div>
`;
